app/assets/javascripts/discourse/app/lib/plugin-api.js

```diff
@@ -2559,6 +2562,24 @@ class PluginApi {
   addBulkActionButton(opts) {
     _addBulkButton(opts);
   }
+
+  /**
+   * Allows you add additional search results to the full page search results.
+   * Search results added through this API will be mixed with initial search results
+   * and ranked using the reciprocal ranking fusion algorithm.
+   *
+   * (See `reciprocalRankingAlgorithm()` in `discourse/app/lib/search.js` for more
+   * details on how reciprocal ranking works)
+   *
+   * Full page must be refreshed using `recheckSearchResults()` to see the new results.
+   *
+   * Example:
+   *
+   * api.addAdditionalSearchResults(newPostArray);
+   */
+  addAdditionalSearchResults(results) {
```

Comment:
Using `add` and `Additional` doesn't sound quite right, it's a little repetitive. Maybe there's a better name for this?

Reply:
Ah yes I agree. How about one of the these:
- `addSearchResults`
- `addCustomSearchResults`
- `addMoreSearchResults`
- `addToSearchResults`

Or any other suggestion you may have? I'm not particular tied to any of them.

Reply:
`addSearchResults` is probably my favourite from the list.